### PR TITLE
Update Terraria Vanilla variables

### PR DESCRIPTION
**Pull request by Braastos**
# Description

Added Password variable, so a password could be easily used
Added World Seed, so that creating a world with a specific world seed would be possible
added NPCStream , to manage bandwidth usage
also updated the wiki link to use the new official wiki (terraria.wiki.gg)

## Checklist for all submissions

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [X] Have you tested and reviewed your changes with confidence that everything works?
* [X] Did you branch your changes and PR from that branch and not from your master branch?

### CODE REVIEW

File: game_eggs/terraria/vanilla/egg-terraria-vanilla.json [hunk 1]
```diff
@@ -1,41 +1,42 @@
...
...p \"${CLEAN_VERSION}\" | cut -d'?' -f1)\r\nfi \r\n\r\n## this is a simple script to validate a download url actaully exists\r\nif [ ! -z \"${DOWNLOAD_LINK}\" ]; then \r\n    if curl --output \/dev\/null --silent --head --fail ${DOWNLOAD_LINK}; then\r\n        echo -e \"link is valid.\"\r\n    else        \r\n        echo -e \"link is invalid closing out\"\r\n        exit 2\r\n    fi\r\nfi\r\n\r\nCLEAN_VERSION=$(echo ${DOWNLOAD_LINK##*\/} | cut -d'-' -f3 | cut -d'.' -f1)\r\n\r\necho -e \"running 'curl -sSL ${DOWNLOAD_LINK} -o ${DOWNLOAD_LINK##*\/}'\" \r\ncurl -sSL ${DOWNLOAD_LINK} -o ${DOWNLOAD_LINK##*\/}\r\n\r\necho -e \"Unpacking server files\"\r\nunzip ${DOWNLOAD_LINK##*\/}\r\n\r\necho -e \"\"\r\ncp -R ${CLEAN_VERSION}\/Linux\/* .\/\r\nchmod +x TerrariaServer.bin.x86_64\r\n\r\necho -e \"Cleaning up extra files.\"\r\nrm -rf ${CLEAN_VERSION}\r\n\r\necho -e \"Generating config file\"\r\ncat <<EOF > serverconfig.txt\r\nworldpath=\/home\/container\/saves\/Worlds\r\nworldname=default\r\nworld=\/home\/container\/saves\/Worlds\/default.wld\r\ndifficulty=3\r\nautocreate=1\r\nport=7777\r\nmaxplayers=8\r\nEOF\r\n\r\nmkdir -p \/mnt\/server\/saves\/Worlds\r\n\r\necho -e \"Install complete\"",
```
Comment (parkervcp) [inline game_eggs/terraria/vanilla/egg-terraria-vanilla.json:25]: The wiki.gg is their official wiki and I see no reason to change the link

File: game_eggs/terraria/vanilla/egg-terraria-vanilla.json [hunk 2]
```diff
@@ -80,7 +85,38 @@
...
+            "default_value": "",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "nullable|string",
+            "field_type": "text"
+        },
+        {
+            "name": "NPCStream",
+            "description": "Reduces enemy skipping but increases bandwidth usage. The lower the number the less skipping will happen, but more data is sent. 0 is off.",
+            "env_variable": "NPCSTREAM",
```
Comment (parkervcp) [inline game_eggs/terraria/vanilla/egg-terraria-vanilla.json:114]: What is the default value for this? So far as I can tell you are just forcing a value that drives higher bandwidth.

**Review by parkervcp — CHANGES_REQUESTED** [review]

File: game_eggs/terraria/vanilla/egg-terraria-vanilla.json (hunk 1 above)
Comment (parkervcp) [reply game_eggs/terraria/vanilla/egg-terraria-vanilla.json:25]: I see I read that backwards. when I cop and pasted the diff.

File: game_eggs/terraria/vanilla/egg-terraria-vanilla.json (hunk 2 above)
Comment (Braastos) [reply game_eggs/terraria/vanilla/egg-terraria-vanilla.json:114]: The default value should be 60 (thats the value which is set in an default serverconfig.txt, but is commented out) but will result in bad enemy lagging, from my experience.
A value of 0 resulted, in my testing, with 1 Player connected to the server, to a maximum bandwidth usage of about 8 kB/s outbound traffic, inbound traffic did not change noticable.
Thats the reason why i defaulted to 0

Comment (parkervcp) [reply game_eggs/terraria/vanilla/egg-terraria-vanilla.json:114]: That's reasonable

**Review by parkervcp — APPROVED** [review]

**Pull request merged**